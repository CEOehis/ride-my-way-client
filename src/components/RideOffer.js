import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import parseTime from '../utils/parseTime';

const RideOffer = ({ ride }) => {
  const {
    origin,
    destination,
    rideCreator,
    departureDate,
    departureTime,
    seats,
    rideId,
  } = ride;
  return (
    <div className="available-ride">
      <div className="path">
        <h4>
          {origin}
          <span>&gt;&gt;</span>
          {destination}
        </h4>
      </div>
      <div className="ride-summary">
        <div className="profile">
          <img src="https://placehold.it/100/100" alt="driver" />
          <div className="driver">
            <h4>{rideCreator}</h4>
            <p>
              {moment(departureDate).format('MMM DD, YYYY')}
              ,
              {parseTime(departureTime)}
            </p>
          </div>
        </div>
        <div className="ride-info">
          <p className="seats">
            Available seats:
            {seats}
          </p>
        </div>
        <div className="actions">
          <a href={`/ride/${rideId}`} className="btn btn-orange btn-lg">Details</a>
        </div>
      </div>
    </div>
  );
};

RideOffer.propTypes = {
  ride: PropTypes.shape({
    origin: PropTypes.string.isRequired,
    destination: PropTypes.string.isRequired,
    rideCreator: PropTypes.string.isRequired,
    departureDate: PropTypes.string.isRequired,
    departureTime: PropTypes.string.isRequired,
    seats: PropTypes.number.isRequired,
    rideId: PropTypes.number.isRequired,
  }).isRequired,
};

export default RideOffer;
