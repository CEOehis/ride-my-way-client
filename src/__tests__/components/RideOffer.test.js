import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RideOffer from '../../components/RideOffer';

Date.now = jest.fn(() => 1538380166917);

const ride = {
  createdAt: '2018-08-26T17:08:27.287Z',
  departureDate: Date.now(),
  departureTime: '12:30:00+01',
  destination: 'Port Liliane',
  origin: 'Reichelchester',
  rideCreator: 'Celestine Ekoh-Ordan',
  rideId: 1,
  seats: 2,
  updatedAt: '2018-08-26T17:08:27.287Z',
};

describe('Form submit RideOffer component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<RideOffer ride={ride} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
