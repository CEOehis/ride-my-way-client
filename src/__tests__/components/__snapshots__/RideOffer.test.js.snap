// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form submit RideOffer component should render correctly 1`] = `
<div
  className="available-ride"
>
  <div
    className="path"
  >
    <h4>
      Reichelchester
      <span>
        &gt;&gt;
      </span>
      Port Liliane
    </h4>
  </div>
  <div
    className="ride-summary"
  >
    <div
      className="profile"
    >
      <img
        alt="driver"
        src="https://placehold.it/100/100"
      />
      <div
        className="driver"
      >
        <h4>
          Celestine Ekoh-Ordan
        </h4>
        <p>
          Oct 01, 2018
          ,
          12:30 AM
        </p>
      </div>
    </div>
    <div
      className="ride-info"
    >
      <p
        className="seats"
      >
        Available seats:
        2
      </p>
    </div>
    <div
      className="actions"
    >
      <a
        className="btn btn-orange btn-lg"
        href="/ride/1"
      >
        Details
      </a>
    </div>
  </div>
</div>
`;
